Assets/Scripts/Utils.cs

```diff
@@ -1,38 +1,47 @@
 ﻿using UnityEngine;
-
-using System;
 using System.Collections.Generic;
+using System.Linq;
 using Random = UnityEngine.Random;
 
-class WeightedRandomBag<T>
+public class WeightedRandomBag<T>
 {
     private struct Entry
     {
-        public float accumulatedWeight;
-        public T item;
+        public float AccumulatedWeight;
+        public T Item;
     }
 
-    private List<Entry> entries = new List<Entry>();
+    private readonly List<Entry> entries = new List<Entry>();
     private float accumulatedWeight;
 
     public void AddEntry(T item, float weight)
     {
         accumulatedWeight += weight;
-        entries.Add(new Entry {item = item, accumulatedWeight = accumulatedWeight});
+        entries.Add(new Entry {Item = item, AccumulatedWeight = accumulatedWeight});
     }
 
     public T GetRandom()
     {
         var r = Random.value * accumulatedWeight;
 
-        foreach (var entry in entries)
+        foreach (var entry in entries.Where(entry => entry.AccumulatedWeight >= r))
         {
-            if (entry.accumulatedWeight >= r)
-            {
-                return entry.item;
-            }
+            return entry.Item;
         }
 
-        return default(T); //should only happen when there are no entries
+        return default; //should only happen when there are no entries
+    }
+}
+
+public static class AudioSourceExtension
+{
+    public static AudioSource PlayClipAndGetSource(AudioClip clip, Vector3 pos){
```

Comment:
What is this for...?

Reply:
It's essentially the same thing as `AudioSource.PlayClipAtPoint()`, except it returns the `AudioSource` used to play it. This way you can change the properties of the `AudioSource` if the defaults aren't what you want.